File: src/mpmt_mss/client/python/led_example.py
from mssclient import MSSClient
from mpmt_mss.feb.devices import DeviceType

CHANNEL_TEST = 7

client = MSSClient("http://zynq:8000/rpc")

print(client.febmgr.getDefinedChannels())
print(client.febmgr.getStatus(DeviceType.PMT))
print(client.febmgr.getStatus(DeviceType.LED))

print(client.febmgr.getLEDStatus(CHANNEL_TEST))
print(client.febmgr.getLEDInfo(CHANNEL_TEST))

print(client.febmgr.getLEDTriggerStatus(CHANNEL_TEST))
print(client.febmgr.getLEDBiasStatus(CHANNEL_TEST))

print(client.febmgr.getLEDBiasVoltage(CHANNEL_TEST))
print(client.febmgr.readLEDBiasVoltage(CHANNEL_TEST))

print(client.febmgr.getLEDTriggerSource(CHANNEL_TEST))

print(client.febmgr.getLEDCurrent(CHANNEL_TEST))
print(client.febmgr.getLEDChannels(CHANNEL_TEST))

print(client.febmgr.readLEDMonRegisters(CHANNEL_TEST))

client.febmgr.powerLEDOn(CHANNEL_TEST)
client.febmgr.powerLEDOff(CHANNEL_TEST)

client.febmgr.setLEDTrigger(CHANNEL_TEST, 1)
client.febmgr.setLEDTriggerSource(CHANNEL_TEST, 2)

client.febmgr.setLEDBias(CHANNEL_TEST, 1)
client.febmgr.setLEDBiasVoltage(CHANNEL_TEST, 5.5)

client.febmgr.setLEDChannels(CHANNEL_TEST, channels=[1, 2])
client.febmgr.setLEDChannels(CHANNEL_TEST, channels=[3], append=True)

File: src/mpmt_mss/client/cpp/examples/led_example.cpp
#include <iostream>

#include <mpmt_mss/mssclient.hpp>

// C++ port of client/python/led_example.py

using namespace mpmt_mss;

constexpr int kChannelTest = 7;

static void print_channels(const std::vector<int>& channels) {
  std::cout << "[";
  for (size_t i = 0; i < channels.size(); i++) std::cout << (i ? ", " : "") << channels[i];
  std::cout << "]" << std::endl;
}

int main() {
  MSSClient client("http://zynq:8000/rpc");

  print_channels(client.febmgr.getDefinedChannels());
  std::cout << client.febmgr.getStatus(DeviceType::PMT).dump() << std::endl;
  std::cout << client.febmgr.getStatus(DeviceType::LED).dump() << std::endl;

  std::cout << client.febmgr.getLEDStatus(kChannelTest).dump() << std::endl;
  std::cout << client.febmgr.getLEDInfo(kChannelTest).dump() << std::endl;

  std::cout << client.febmgr.getLEDTriggerStatus(kChannelTest).dump() << std::endl;
  std::cout << client.febmgr.getLEDBiasStatus(kChannelTest).dump() << std::endl;

  std::cout << client.febmgr.getLEDBiasVoltage(kChannelTest) << std::endl;
  std::cout << client.febmgr.readLEDBiasVoltage(kChannelTest) << std::endl;

  std::cout << client.febmgr.getLEDTriggerSource(kChannelTest).dump() << std::endl;

  std::cout << client.febmgr.getLEDCurrent(kChannelTest) << std::endl;
  print_channels(client.febmgr.getLEDChannels(kChannelTest));

  std::cout << client.febmgr.readLEDMonRegisters(kChannelTest).dump() << std::endl;

  client.febmgr.powerLEDOn(kChannelTest);
  client.febmgr.powerLEDOff(kChannelTest);

  client.febmgr.setLEDTrigger(kChannelTest, true);
  client.febmgr.setLEDTriggerSource(kChannelTest, TriggerSource::EXT);

  client.febmgr.setLEDBias(kChannelTest, true);
  client.febmgr.setLEDBiasVoltage(kChannelTest, 5.5);

  client.febmgr.setLEDChannels(kChannelTest, {1, 2});
  client.febmgr.setLEDChannels(kChannelTest, {3}, /*append=*/true);

  return 0;
}
